src/main/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitChecker.java

```diff
@@ -107,7 +107,18 @@ public void checkApiRateLimit(@NonNull TaskListener listener, GitHub github) thr
                 waitUntilRateLimit(listener, github, rateLimit, expiration);
             }
         }
-    };
+    },
+    /**
+     * Ignore GitHub API Rate limit. Useful for GitHub Enterprise instances that might not have a limit set up.
+     */
+    NoThrottle(Messages.ApiRateLimitChecker_NoThrottle()) {
+
+                @Override
+                public void checkApiRateLimit(@NonNull TaskListener listener, GitHub github) throws IOException, InterruptedException {
+                    // Nothing needed
```

Comment:
If pointing at `https://github.com` this setting should pass through to `ThrottleOnOver`.  There is literally never a reason to not check rate limits when interacting with `github.com`.

Reply:
Should we block the selection of `NoThrottle` in the config page if the api uri is github.com?